microcosm_pubsub/result.py

```diff
@@ -154,8 +154,18 @@ def log(self, logger, opaque):
         )
 
     def error_reporting(self, sentry_enabled, opaque):
-        if not sentry_enabled:
+        if any([
+            not sentry_enabled,
+            self.result not in [
+                MessageHandlingResultType.FAILED,
+                MessageHandlingResultType.EXPIRED,
+            ],
+            not self.exc_info,
+        ]):
```

Comment:
I think this'd be clearer if you demorgan out the `not` to a `if not all([...])` instead.